NDTensors/src/lib/UnallocatedArrays/src/abstractfill/abstractfill.jl

```diff
@@ -9,7 +9,7 @@ end
 
 ## TODO this fails if the parameter is a type
 function alloctype(Atype::Type{<:AbstractFill})
-  return get_parameter(Atype, Position{4}())
+  return parameter(Atype, 4)
 end
 
-set_axestype(T::Type{<:AbstractFill}, ax::Type) = set_parameters(T, Position{3}(), ax)
+set_axestype(T::Type{<:AbstractFill}, ax::Type) = set_parameter(T, 3, ax)
```

Comment:
I think for type stability we need to go back to using `Position(3)`. Same for other calls in the code.

Reply:
I was thinking we could use the 
```
position(::Type{<:AbstractFill}, ::typeof(axes)) = Position(3)
set_axestype(T::Type{<:AbstractFill}, ax::Type) = set_parameter(T, position(T, axes), ax)
```
Kind of functionality for this. I will keep this open because I need to change it still in the code. Thanks for adding this comment for a good reminder

Reply:
Sounds like a good plan. I guess it could be even more general:
```julia
set_axestype(type::Type, parameter) = set_parameter(type, position(type, axestype), parameter)
```
(also it seems like it should be based on `axestype`, not `axes`).

Reply:
Just as a quick update, I realized I can actually do 
```
set_axestype(type::Type, parameter) = set_parameter(type, axestype, parameter)
```
But I am currently working on this kind of code and looking through the whole code base

Reply:
This system seems to work across the board and I implemented it for `specify_parameter` as well!